### PR TITLE
Adds BNF and text to extend LANGTAG (now LANG_DIR) to support base direction

### PR DESCRIPTION
**Pull request by gkellogg**
* Adds BNF and text to change LANGTAG to LANG_DIR to support base direction and term constructors for creating directional language-tagged strings.
* Update spec/turtle.bnf

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-turtle/pull/49.html" title="Last updated on Nov 3, 2023, 7:56 PM UTC (e22eec5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-turtle/49/5ef92ec...e22eec5.html" title="Last updated on Nov 3, 2023, 7:56 PM UTC (e22eec5)">Diff</a>

### CODE REVIEW

File: spec/index.html [hunk 1]
```diff
@@ -124,7 +124,9 @@
     <a data-cite="RDF12-CONCEPTS#dfn-subject">subject</a> or
     <a data-cite="RDF12-CONCEPTS#dfn-object">object</a> of another
     <a data-cite="RDF12-CONCEPTS#dfn-rdf-triple">triple</a>,
-    making it possible to make statements about other statements.</p>
+    making it possible to make statements about other statements.
+    RDF 1.2 Turtle also adds support for
+    <a data-cite="RDF12-CONCEPTS#dfn-dir-lang-string">directional language-tagged strings</a>.</p>
 
   <p>RDF 1.2 Turtle also introduces an <a>annotation syntax</a> which allows
```
Comment (TallTed) [inline spec/index.html:131]:
```suggestion
  <p>In addition, RDF 1.2 Turtle introduces an <a>annotation syntax</a> which allows
```

File: spec/index.html [hunk 2]
```diff
@@ -506,6 +516,7 @@ <h3>Quoted Literals</h3>
         show:218 rdfs:label "That Seventies Show"^^<http://www.w3.org/2001/XMLSchema#string> . # same as above
         show:218 rdfs:label "That Seventies Show" .                                            # same again
         show:218 show:localName "That Seventies Show"@en .                 # literal with a language tag
+        show:218 show:localName "That Seventies Show"@en-ltr .             # literal with a language tag and base direction
```
Comment (TallTed) [inline spec/index.html:519]:
```suggestion
        show:218 show:localName "That Seventies Show"@en--ltr .             # literal with a language tag and base direction
```

File: spec/index.html [hunk 3]
```diff
@@ -506,6 +516,7 @@ <h3>Quoted Literals</h3>
...
         show:218 show:localName 'Cette Série des Années Soixante-dix'@fr . # literal delimited by single quote
         show:218 show:localName "Cette Série des Années Septante"@fr-be .  # literal with a region subtag
```
Comment (TallTed) [inline spec/index.html:521]: Might also want to add an example with both regional sub-tag and base direction, like `@fr-be--ltr`?

File: spec/index.html [hunk 4]
```diff
@@ -1498,7 +1511,10 @@ <h2>Parsing</h2>
     <a data-cite="RDF12-CONCEPTS#dfn-blank-node">blank nodes</a>, and
     <a data-cite="RDF12-CONCEPTS#dfn-quoted-triple">quoted triples</a>.
 
-    Literals are composed of a <a data-cite="RDF12-CONCEPTS#dfn-lexical-form">lexical form</a> and an optional <a data-cite="RDF12-CONCEPTS#dfn-language-tag">language tag</a> [[!BCP47]] or datatype IRI.
+    Literals are composed of a <a data-cite="RDF12-CONCEPTS#dfn-lexical-form">lexical form</a>
+    and an optional <a data-cite="RDF12-CONCEPTS#dfn-language-tag">language tag</a> [[!BCP47]]
+    – possibly including a <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a> –
+    or <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a>.
```
Comment (TallTed) [inline spec/index.html:1517]:
```suggestion
    or an optional <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a>.
```

File: spec/index.html [hunk 5]
```diff
@@ -1595,8 +1611,8 @@ <h3>RDF Term Constructors</h3>
...
...;"            ><a class="type literal"      href="#grammar-production-RDFLiteral"                      >RDFLiteral                      </a></td><td><a data-cite="RDF12-CONCEPTS#dfn-literal">      literal      </a></td><td>The literal has a lexical form of the first rule argument, <a href="#grammar-production-String"><code>String</code></a>. If the <code>'^^' iri</code> rule matched, the <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a> is <code>iri</code> and the literal has no language tag. If the <a href="#grammar-production-LANG_DIR" class="type langDir"><code>LANG_DIR</code></a> rule matched, the <a data-cite="RDF12-CONCEPTS#dfn-language-tag">language tag</a> and <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a> are taken from <a href=#handle-LANG_DIR class="type langDir">LANG_DIR</a>. If there is no <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a>, the datatype is <code>rdf:langString</code>. If there is a <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a>, the datatype is <code>rdf:dirLangString</code>. If neither matched, the datatype is <code>xsd:string</code> and the literal has no language tag.</td></tr>
```
Comment (TallTed) [inline spec/index.html:1615]: `the <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a> is <code>iri</code>` seems to require that the datatype IRI actually *be* an IRI, which `iri` is not. I *think* this should be `xsd:anyURI`, so (I notice that GitHub doesn't properly highlight this tiny change 😢 ) —
```suggestion
      <tr id="handle-RDFLiteral"                      ><td style="text-align:left;"            ><a class="type literal"      href="#grammar-production-RDFLiteral"                      >RDFLiteral                      </a></td><td><a data-cite="RDF12-CONCEPTS#dfn-literal">      literal      </a></td><td>The literal has a lexical form of the first rule argument, <a href="#grammar-production-String"><code>String</code></a>. If the <code>'^^' iri</code> rule matched, the <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a> is <code>xsd:anyURI</code> and the literal has no language tag. If the <a href="#grammar-production-LANG_DIR" class="type langDir"><code>LANG_DIR</code></a> rule matched, the <a data-cite="RDF12-CONCEPTS#dfn-language-tag">language tag</a> and <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a> are taken from <a href=#handle-LANG_DIR class="type langDir">LANG_DIR</a>. If there is no <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a>, the datatype is <code>rdf:langString</code>. If there is a <a data-cite="RDF12-CONCEPTS#dfn-base-direction">base direction</a>, the datatype is <code>rdf:dirLangString</code>. If neither matched, the datatype is <code>xsd:string</code> and the literal has no language tag.</td></tr>
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: spec/index.html (hunk 5 above)
Comment (gkellogg) [reply spec/index.html:1615]: I'm going to say that the `datatype IRI` is _derived_ from `iri`, as this is about constructing an RDF Literal, which uses an IRI not `xsd;anyUri`. I don't think we need to spell out here how that parsed value is turned into an IRI, as we do so elsewhere.

**Pull request merged**